### DISCUSSION TITLE
Documenting creation of a sub-material theme

### DISCUSSION THREAD

**Discussion by ssbarnea** [squidfunk/mkdocs-material#4918] [General]
With sphinx it was easy to extend an existing theme and build another one on top of it, publishing it on pypa for easy use. Still, I was not able to find any information regarding doing something similar for mkdocs-material.

I should probably mention that I am **not** looking for a forking approach, as that would require a lot of maintenance. I am more interested in writing a publishing a theme that just extends mkdocs-material with stuff like specific icons, colors and other configuration that you might want to keep consistent across multiple project repositories, and avoid having to duplicate these settings on all the projects.

Thinks I identified so far:

- custom css
- favicon and logo
- theme palette
- possibly even the default list of plugins or extensions

**Comment by squidfunk** [comment]
Yes, that's possible. See:

- https://www.mkdocs.org/dev-guide/themes/#extends
- #4901#discussioncomment-4761396

**Reply by squidfunk** [reply]
... by the way, I'm happy to include such a guide in the documentation! It would be perfect, if you could document your process of creating a sub theme, since this has been asked several times, and I haven't had the time to write something up. Additionally, getting a user perspective on this might make more sense. It would be a great contribution to the project!

**Reply by ssbarnea** [reply]
I will try it and contribute few words once I do it. Basically I was asking for an example of using extends as I can extrapolate from one. If none exists, i can attempt to make one as the use-case is very simple and I want to keep it very simple.
